operator/v1/types_network.go

```diff
@@ -428,7 +429,15 @@ type HybridOverlayConfig struct {
 	HybridOverlayVXLANPort *uint32 `json:"hybridOverlayVXLANPort,omitempty"`
 }
 
+// +kubebuilder:validation:XValidation:rule="self == oldSelf || has(self.mode)",message="ipsecConfig.mode is required"
 type IPsecConfig struct {
+	// mode defines the ipsec configuration level, 'Disabled' disable ipsec on the node level,
+	// 'External' enables ipsec on the node level but requires the user to configure it (mostly for externl
+	// communication), and 'Full' enables ipsec on the node level and configures it for secure communciation
+	// between pods in the cluster (the user needs to configure it for external secure communciation).
+	// +kubebuilder:validation:Enum=Disabled;External;Full
+	// +optional
+	Mode IPsecMode `json:"mode,omitempty"`
```

Comment:
I'd like to improve the wording on this, though I don't quite have enough context to finish this, PTAL
```suggestion
	// mode defines the behaviour of the ipsec configuration within the platform.
	// Valid values are `Disabled`, `External`, `Full` and omitted.
	// When 'Disabled', ipsec will not be configured at the node level.
	// When 'External', ipsec is configured on the node level but requires the user to configure <what>?.
	// When 'Full', ipsec is configured on the node level and inter-pod secure communication within the cluster is configured.
	// Note with `Full`, if ipsec is desired for inter-node communication, this is left to the user to configure.
	// When omitted, this means no opinion and the platform is left to choose a reasonable default which is
	// subject to change over time.
	// The current default behaviour is `Full`.
	// +kubebuilder:validation:Enum=Disabled;External;Full
	// +optional
	Mode IPsecMode `json:"mode,omitempty"`
```

What does it actually mean for ipsec to be configured? Does it mean ports are exposed? Can you explain it to me in simple terms?

Reply:
ipsec configuration is quite complex, it includes the encryption protocol the keys and policies which decides which communications is encrypted and which is not. There are around 20 configuration parameters that we added to nmstate in order to support the ipsec configuration, and this is a subset of all the configuration options in libreswan.